tests/pytests/test_empty.py

```diff
@@ -888,3 +681,573 @@ def testEmptyInfo():
+            '}',
+            ''
+        ]
+        env.assertEqual(res, expected)
+
+        # INTERSECTION operator for a single TEXT field
+        res = env.cmd('FT.EXPLAINCLI', 'idx', '@text:(foo "")')
+        expected = [
+            '@text:INTERSECT {',
+            '  @text:UNION {',
+            '    @text:foo',
+            '    @text:+foo(expanded)',
+            '  }',
+            '  @text:""',
+            '}',
+            ''
+        ]
+        env.assertEqual(res, expected)
+
+        res = env.cmd('FT.EXPLAINCLI', 'idx', '@text:("" bar)')
+        expected = [
+            '@text:INTERSECT {',
+            '  @text:""',
+            '  @text:UNION {',
+            '    @text:bar',
+            '    @text:+bar(expanded)',
+            '  }',
+            '}',
+            ''
+        ]
+        env.assertEqual(res, expected)
+
+        res = env.cmd('FT.EXPLAINCLI', 'idx', '@text:("" "")')
+        expected = [
+            '@text:INTERSECT {',
+            '  @text:""',
+            '  @text:""',
+            '}',
+            ''
+        ]
+        env.assertEqual(res, expected)
+
+def testInvalidUseOfEmptyString(env):
+    """Tests that invalid syntax for empty values is rejected by the parser"""
+
+    MAX_DIALECT = set_max_dialect(env)
+    for dialect in range(2, MAX_DIALECT + 1):
+        env = Env(moduleArgs="DEFAULT_DIALECT " + str(dialect))
+        conn = getConnectionByEnv(env)
+
+        dim = 4
+        # Create an index
+        env.expect(
+            'FT.CREATE', 'idx', 'SCHEMA',
+            't', 'TAG', 'INDEXEMPTY',
+            'text', 'TEXT', 'INDEXEMPTY', 'PHONETIC', 'dm:en',
+            'location', 'GEO',
+            'v', 'VECTOR', 'FLAT', '6', 'DIM', dim, 'DISTANCE_METRIC', 'L2',
+            'TYPE', 'FLOAT32',).ok()
+        conn.execute_command('HSET', 'h1', 't', '')
+        conn.execute_command('HSET', 'h2', 't', 'a')
+
+        # Unsupported empty string in fuzzy terms
+        res = conn.execute_command('FT.SEARCH', 'idx', '@text:(%""%)')
+        env.assertEqual(res, EMPTY_RESULT)
+
+        res = conn.execute_command('FT.SEARCH', 'idx', '@text:(%%""%%)')
+        env.assertEqual(res, EMPTY_RESULT)
+
+        res = conn.execute_command('FT.SEARCH', 'idx', '@text:(%%%""%%%)')
+        env.assertEqual(res, EMPTY_RESULT)
+
+        res = conn.execute_command('FT.SEARCH', 'idx', '@text:(%$p%)', 
+                                   'PARAMS', 2, 'p', '""')
+        env.assertEqual(res, EMPTY_RESULT)
+
+        res = conn.execute_command('FT.SEARCH', 'idx', '@text:(%%$p%%)', 
+                                   'PARAMS', 2, 'p', '""')
+        env.assertEqual(res, EMPTY_RESULT)
+
+        res = conn.execute_command('FT.SEARCH', 'idx', '@text:(%%%$p%%%)', 
+                                   'PARAMS', 2, 'p', '""')
+        env.assertEqual(res, EMPTY_RESULT)
+        
+        # Invalid use of empty string in geo filter
+        env.expect('FT.SEARCH', 'idx', '@location:[1.23 4.56 10 ""]').error().\
```

Comment:
Also test using empty params?

Reply:
Maybe in `testEmptyParam`

Reply:
Also for query attribute values, e.g., 
```
'@t:{abc}=>{$weight:$param}'
```

Reply:
There are already such tests, e.g.,:
```
env.expect('FT.SEARCH', 'idx', '@t:{abc}=>{$weight:$p}',
                   'PARAMS', 2, 'p', '').error().contains(expected_error)
```
[Here](https://github.com/RediSearch/RediSearch/blob/f4b93bef41d9e68b1097db1a5ec76257b991ade6/tests/pytests/test_empty.py#L1212-L1213).